.azure/pipelines/ci.yml

```diff
@@ -7,22 +7,661 @@ pr:
+              /bl:artifacts/logs/SharedFx-Installers.binlog
+      displayName: Build SharedFX Installers
+    - bash: docker system prune -af
+      displayName: Docker prune
+    - task: PublishBuildArtifacts@1
+      displayName: Upload artifacts
+      condition: eq(variables['system.pullrequest.isfork'], false)
+      continueOnError: true
+      inputs:
+        pathtoPublish: artifacts/
+        artifactName: artifacts-SharedFx-Installers
+        artifactType: Container
+        parallel: true
+
+  - job: Publish
+    displayName: Publish
+    dependsOn: 
+      - Windows_Installers
+      - SharedFX_Installers
+      - Package_Archive
+    timeoutInMinutes: 90
+    workspace:
+      clean: all
+    pool:
+      name: NetCoreInternal-Pool
+      queue: BuildPool.Server.Amd64.VS2017
+    variables:
+      _SignType: real
+      JAVA_HOME: $(Agent.BuildDirectory)\.tools\jdk
+    steps:
+    - checkout: self
+      clean: true
+    - task: NuGetCommand@2
+      displayName: 'Clear NuGet caches'
+      condition: succeeded()
+      inputs:
+        command: custom
+        arguments: 'locals all -clear'
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Windows artifacts
+      inputs:
+        artifactName: artifacts-Windows-Release
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Linux SharedFx artifacts
+      inputs:
+        artifactName: artifacts-Linux-SharedFx
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Linux Musl SharedFx artifacts
+      inputs:
+        artifactName: artifacts-Linux-Musl-SharedFx
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download MacOs SharedFx artifacts
+      inputs:
+        artifactName: artifacts-MacOs-SharedFx
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Windows SharedFx artifacts
+      inputs:
+        artifactName: artifacts-Windows-SharedFx
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Package Archive artifacts
+      inputs:
+        artifactName: artifacts-Package-Archive
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download SharedFx installer artifacts
+      inputs:
+        artifactName: artifacts-SharedFx-Installers
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: DownloadBuildArtifacts@0
+      displayName: Download Windows installer artifacts
+      inputs:
+        artifactName: artifacts-Windows-Installers
+        downloadPath: $(Build.StagingDirectory)/downloaded_artifacts/
+    - task: CopyFiles@2
```

Comment:
This giant block of CopyFile tasks is really ugly, but I couldn't get it to work with fewer - @riarenas @mmitche is there a cleaner way to copy all of the inputs without doing them one-by-one?